### PR TITLE
add com.jonathankang.Weibird

### PR DESCRIPTION
**Pull request by JonathanKang**

### CODE REVIEW

File: com.jonathankang.Weibird.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+    "app-id" : "com.jonathankang.Weibird",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "master",
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: Flathub targets stable releases – thus also requires using stable runtimes and project releases.

File: com.jonathankang.Weibird.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "weibird",
+    "desktop-file-name-prefix" : "(Master)",
+    "finish-args" : [
+        "--device=dri",
+        "--share=ipc",
+        "--share=network",
+        "--socket=wayland",
+        "--socket=x11",
+        "--filesystem=host",
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: What does it need host access for?

File: com.jonathankang.Weibird.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
+        "cflags" : "-O0 -g"
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: Build flags are automatically set by runtime these days.

File: com.jonathankang.Weibird.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    "cleanup" : [
+        "/include",
+        "/lib/pkgconfig"
+    ],
+    "modules" : [
+        {
+            "name" : "meson",
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: It's already part of the runtime.

File: com.jonathankang.Weibird.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                    "branch" : "librest-0-7"
+                }
+            ]
+        },
+        {
+            "name" : "weibird",
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "build-options" : {
+                "build-args" : ["--share=network"]
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: We don't allow network access during build.

File: com.jonathankang.Weibird.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                    "type"   : "git",
+                    "url"    : "https://gitlab.gnome.org/GNOME/librest",
+                    "branch" : "librest-0-7"
+                }
+            ]
+        },
+        {
+            "name" : "weibird",
+            "buildsystem" : "meson",
+            "builddir" : true,
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: `builddir` is turned on by default for meson buildsystem.

File: com.jonathankang.Weibird.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        },
+        {
+            "name" : "librest",
+            "buildsystem" : "autotools",
+            "builddir" : true,
+            "sources" : [
+                {
+                    "type"   : "git",
+                    "url"    : "https://gitlab.gnome.org/GNOME/librest",
+                    "branch" : "librest-0-7"
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: Please use stable release.

File: com.jonathankang.Weibird.json [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "build-options" : {
+                "build-args" : ["--share=network"]
+            },
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/JonathanKang/weibird",
+                    "branch" : "master"
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: Please use a stable tag or specific commit.

File: com.jonathankang.Weibird.json [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                {
+                    "type" : "archive",
+                    "url" : "https://files.pythonhosted.org/packages/df/df/8d5c006f87a3c75f588aeadcf089726bba1e567997a1088655fe7ed68f3d/meson-0.48.0.tar.gz",
+                    "sha256" : "ecc6b319d86362d1ae2a64c40738f15dff5921b01cdba5d6506c877e4dd55c61"
+                }
+            ]
+        },
+        {
+            "name" : "librest",
+            "buildsystem" : "autotools",
```
Comment (barthalion) [inline com.jonathankang.Weibird.json]: autotools is the default build system (but it's a nitpick really)

File: com.jonathankang.Weibird.json (hunk 1 above)
Comment (JonathanKang) [reply com.jonathankang.Weibird.json]: So it's not allowed ot distribute Apps which are build from master branch, right?

Comment (barthalion) [reply com.jonathankang.Weibird.json]: We allow beta/RC releases of project themselves, but yes, we require stable runtime.

I see you tagged some releases already: https://github.com/JonathanKang/weibird/releases
Do they fail to build against 3.30 runtime?

Comment (JonathanKang) [reply com.jonathankang.Weibird.json]: No. I've fixed it using those tag release. Thanks.

File: com.jonathankang.Weibird.json (hunk 9 above)
Comment (JonathanKang) [reply com.jonathankang.Weibird.json]: Sure. Removed.

File: com.jonathankang.Weibird.json (hunk 8 above)
Comment (JonathanKang) [reply com.jonathankang.Weibird.json]: Done

**Comment by nedrichards** [comment]
bot, build com.jonathankang.Weibird

**Comment by barthalion** [comment]
Your application either does not provide appdata file or install it at wrong place/filename. You can find appdata guidelines on our wiki: https://github.com/flathub/flathub/wiki/AppData-Guidelines

**Comment by nedrichards** [comment]
bot, build com.jonathankang.Weibird

**Comment by nedrichards** [comment]
Other than a quick install and run of the test build, this lgtm

**Comment by nedrichards** [comment]
bot, build com.jonathankang.Weibird

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.jonathankang.Weibird

**Comment by JonathanKang** [comment]
https://flathub.org/apps/details/com.jonathankang.Weibird show app not found. Is there anything that I need do to fix this?

**Comment by nedrichards** [comment]
I'll have a look and try and build myself later.

**Pull request closed without merging**